mmaction/core/evaluation/average_precision.py

```diff
@@ -0,0 +1,337 @@
+            box = det_result[j, 4:8]
+            is_positive = False
+
+            if key in original_key:
+                if key in gt:
+                    ious = iou2d(gt[key], box)
+                    max_idx = np.argmax(ious)
+
+                    if ious[max_idx] >= threshold:
+                        is_positive = True
+                        gt[key] = np.delete(gt[key], max_idx, 0)
+
+                        if gt[key].size == 0:
+                            del gt[key]
+                    else:
+                        localization_error += 1
+                else:
+                    localization_error += 1
+
+            elif key in other_gt:
+                ious = iou2d(other_gt[key], box)
+                if np.max(ious) >= threshold:
+                    classification_error += 1
+                else:
+                    other_error += 1
+
+            elif label_index in label_dict[key[0]]:
+                time_error += 1
+            else:
+                other_error += 1
+
+            if is_positive:
+                tp += 1
+                fn -= 1
+            else:
+                fp += 1
+
+            precision_recall[i + 1, 0] = tp / max(1, (tp + fp))
+            precision_recall[i + 1, 1] = tp / max(1, (tp + fn))
+            precision_recall[i + 1, 2] = localization_error / max(1, (tp + fp))
+            precision_recall[i + 1,
+                             3] = classification_error / max(1, (tp + fp))
+            precision_recall[i + 1, 4] = time_error / max(1, (tp + fp))
+            precision_recall[i + 1, 5] = other_error / max(1, (tp + fp))
+
+        ap_results.append(pr_to_ap(precision_recall[..., :2]))
+        for j in range(2, 6):
+            other_ap_results[j - 2].append(
+                pr_to_ap(precision_recall[..., [j, 1]]))
+        missing_detections.append(precision_recall[-1, 1])
+
+    ap_results = np.array(ap_results) * 100
+    other_ap_results = np.array(other_ap_results) * 100
+
+    (localization_error, classification_error, time_error,
+     other_error) = other_ap_results[:4]
+    missing_detections = 100 - 100 * np.array(missing_detections)
+
+    result = dict(
+        ap_results=ap_results,
+        localization_error=localization_error,
+        classification_error=classification_error,
+        time_error=time_error,
+        other_error=other_error,
+        missing_detections=missing_detections)
+
+    result_str = ''
+    for i, label in enumerate(labels):
+        result_str += f'{label:20s}' + ' '.join(
+            [f'{v[i]:8.2f}' for v in result.values()]) + '\n'
+    result_str += '\n' + f"{'mean':20s}" + ' '.join(
+        [f'{np.mean(v):8.2f}' for v in result.values()]) + '\n'
+
+    msg = 'Error Analysis\n'
+    msg += f"\n{'label':20s} {'   AP   ':8s} {'  Loc.  ':8s} {'  Cls.  ':8s} "
+    msg += f"{'  Time  ':8s} {' Other ':8s} {' missed ':8s}\n"
+    msg += f'\n{result_str}'
+
+    print(msg)
```

Comment:
Maybe construct the msg string outside this function to avoid several printings due to multi-process.